test/e2e/app-dir/app-client-cache/client-cache.test.ts

```diff
@@ -434,6 +435,51 @@ createNextDescribe(
 
           expect(newNumber).not.toBe(randomNumber)
         })
+
+        it('should respect a loading boundary that returns `null`', async () => {
+          await browser.close()
+          // grab a reference to the original loading file, and patch it with a `null` loading state
+          const originalLoadingFile = await next.readFile('app/[id]/loading.js')
+          await next.stop()
+          await next.patchFile(
+            'app/[id]/loading.js',
+            outdent`
+              export default function Loading() {
+                return null
+              }
+            `
+          )
+
+          await next.start()
```

Comment:
should we use a separate loading file for this to avoid having to re-build for this assertion specifically?